ksql-execution/src/main/java/io/confluent/ksql/execution/plan/TableFilter.java

```diff
@@ -23,14 +25,21 @@
 
 @Immutable
 public class TableFilter<T> implements ExecutionStep<T> {
+  private static final String SOURCE = "source";
+  private static final String FILTER_EXPRESSION = "filterExpression";
+
+  @JsonProperty(PROPERTIES)
   private final ExecutionStepProperties properties;
+  @JsonProperty(SOURCE)
   private final ExecutionStep<T> source;
+  @JsonProperty(FILTER_EXPRESSION)
   private final Expression filterExpression;
 
+  @JsonCreator
   public TableFilter(
-      final ExecutionStepProperties properties,
-      final ExecutionStep<T> source,
-      final Expression filterExpression
+      @JsonProperty(PROPERTIES) final ExecutionStepProperties properties,
+      @JsonProperty(SOURCE) final ExecutionStep<T> source,
```

Comment:
Should we be serializing the whole source execution step? If we have steps a -> b -> c, then when we serialize out b it will also duplicate a, and serializating c will duplicate b and a into the document, right?

Reply:
They won't be duplicated - we'll just have a really deeply nested structure, e.g.:

```
{
    // some stuff
    "source": {
         // some stuff
         "source": {
              ...
         }
     }
}
```
We could flatten it and just serialize the IDs, but it makes the implementation of the data structure more complicated (have to wrap them all in a containing object), and there isn't much gain other than maybe being a bit more human-readable.

Reply:
Gotcha!